Merge of the CyEntEE and EffiziEntEE models

TSteffenTUHH opened the pull request:
Hallo Zusammen,
wie besprochen hier der Pull Request. Ich hab dich, Julian, als Reviewer ausgewählt. Ich hoffe, dass das richtig war.

LG Tom

wibela committed:
Change torque unit from Nm to N.m

wibela committed:
Modify WindTurbine TorqueController (TorqueController_modified) Unit fix

1. Infinite rotor acceleration while low wind speed
Problem: Friction is used to deaccelerate the rotor while low wind speed. But, the friction coefficient is constant and not a function of angular velocity
Solution: Add proportionality factor derived from drag equation

2. Wind turbine does not accelerate with wind speed increase after rotor standstill
Problem: Low angular velocity leads to low cp coefficient. Torque is set in dependency of omega, but is still too high for low speeds, so the torque delivered by rotor is not sufficient. In consequence, too high generator torque after wind speed increase and preceding rotor standstill led to no rotor acceleration.
Solution: If angular velocity is under a certain limit according to its optimal value, the generator is deactivated to let the rotor accelerate.

wibela committed:
recent changes for wind turbine improvements regarding dynamics

wibela committed:
adapt wind turbine changes to new Modelica Version

wibela committed:
small change in start values of EPP of Line Models

wibela committed:
enhanced simulation speed in GenericStorage_Base

if storage is empty and tha battery is being charged with activated stationary loss model and using the model with hysteresis (GenericStorageHyst), the simulation hangs. Deactivating the storage loss model during active hysteresis while storage is empty solves the problem. See changes for detail.
Additionally, the smallest possible number (eps) is used to compare to zero instead.

wibela committed:
enhanced simulation speed in GenericStorage_Base

if storage is empty and tha battery is being charged with activated stationary loss model and using the model with hysteresis (GenericStorageHyst), the simulation hangs. Deactivating the storage loss model during active hysteresis while storage is empty solves the problem. See changes for detail.
Additionally, the smallest possible number (eps) is used to compare to zero instead.

TSteffenTUHH committed:
Extended two pin and resistor model for thermal analysis.

TSteffenTUHH committed:
Small changes in PartialSpecificElement2Pin and Resistor

TSteffenTUHH committed:
Adding LoadResources to GenericDataTable for FMU Standalone Export capability.

wibela committed:
Small bug fix in Heat Pump model

TSteffenTUHH committed:
Revert "enhanced simulation speed in GenericStorage_Base"

This reverts commit b1bd1276d759e3b6e472bcbc2391558ab852bf1c.

wibela committed:
Add Unit SpecificConductance

wibela committed:
Resolve unit warning in PV Plant model

TSteffenTUHH committed:
Added new PVPlantControllable, the corresponding check model and a controlBus. The new PVPlantControllable model is a controllable and complex version of the Advanced_PV/DNIDHI_Input/PVPlant model. Limitations and Setpoints for active and reactive power are set over the included controlBus, which is an expandable interface connector.

TSteffenTUHH committed:
Added SmartMeter model for realistic household metering.

TSteffenTUHH committed:
Added new components to package.orders

TSteffenTUHH committed:
Added Package ElectricVehicle in Consumer/Electrical. The package included multiple characteristics for the onboard charger behavior of non-publicly charged electric vehicles. Additionally A Table_DrivingProfiles model with an example was added. The BEVBattery model was added in Storage/Electrical. Check model for the BatteryElectricVehicle model was added.

TSteffenTUHH committed:
Small bug fix to show Electric Vehicle package in library

TSteffenTUHH committed:
Added first Version of Prosumer. Still work in progress...

TSteffenTUHH committed:
Added better documentation the Electric Vehicle Characteristics.

TSteffenTUHH committed:
Improved Smart Meter -> Conditional delay to be numerically efficient if not applied. Also bug fix in check model.

wibela committed:
Modify Heatpump model

TSteffenTUHH committed:
Bug changes in models pointing to the cyentee package

TSteffenTUHH committed:
Merge branch 'EffiziEntEE' of https://github.com/TransiEnt-official/transient-lib into EffiziEntEE

wibela committed:
Heatpump.Heatpump: bug fixing

wibela committed:
Add new CPP models

wibela committed:
Add new space heating models

wibela committed:
Minimal update on Heatpump model

wibela committed:
Update Prosumer model

include heating
update power boundaries in BEV models
remove storage meter and heating meter

TSteffenTUHH committed:
Bug fixes and consolidation of Prosumer model and submodels for the TransiEnt merge

TSteffenTUHH committed:
Some more bug changes

TSteffenTUHH committed:
Added LV_rural_1 scenarios with data for all scenarios

TSteffenTUHH committed:
Added LV rural 2 scenario

TSteffenTUHH committed:
Added LV semiurban 4 scenario

TSteffenTUHH committed:
Added LV semiurban 5 scenario

TSteffenTUHH committed:
Added LV urban 6 scenario and MV scenarios which are still under test.

TSteffenTUHH committed:
MV Scenarios finished

TSteffenTUHH committed:
Small bug fix

TSteffenTUHH committed:
Another small change since two outputs where forgotten.

urju commented:
Moin @TSteffenTUHH und @wibela, ich habe einen Teil stichprobenartig durchgesehen. Producer, Consumer und Examples folgen noch bis Ende der Woche. Hier sind schon mal meine Notizen:
- [ ] Merge Request enthält 3 .bak-mo-files. Bitte Umgang klären
- [ ] TransiEnt/Basics/Blocks/Check/Check_OnOff.mo ohne Erläuterungen
- [ ] Generell sollten Real/Boolean In- und Outputs mit u bzw. y bezeichnet werden
- [ ] Generell sollten Bezeichnungen wie 'new' im Modellnamen oder 'New features' in der Dokumentationen vermieden werden oder mindestens nicht ohne Datum/Versionsnummer stehen. Insbesondere wird bei den Boundaries nicht klar, was die Entwicklung zu den 'alten' Modellen ist, warum es diese braucht und ob es die alten Modelle trotzdem braucht.
      Beispiele:
	- TransiEnt/Basics/Blocks/PID_reset_lim.mo
	- TransiEnt/Components/Boundaries/Electrical/ComplexPower/PQBoundary_new.mo
	- TransiEnt/Components/Boundaries/Electrical/ComplexPower/PVBoundary_new.mo
	- TransiEnt/Components/Boundaries/Electrical/ComplexPower/PartialBoundary_new.mo
	- TransiEnt/Components/Boundaries/Electrical/ComplexPower/SlackBoundary_new.mo
	- TransiEnt/Components/Boundaries/Electrical/ComplexPower/package.mo
	- TransiEnt/Components/Electrical/Grid/Line_new.mo
- [ ] Zweck von TransiEnt/Basics/Tables/DrivingProfiles/Example_DrivingProfile.mo wird mir nicht klar
- [ ] Wozu TransiEnt/Components/Electrical/Grid/Base/AdmittanceTwoPort.mo im Vergleich zu TransiEnt/Components/Electrical/Grid/Base/PartialTwoPortAdmittanceComplex.mo?
- [ ] TransiEnt/Components/Electrical/Grid/Base/Node.mo eine Ebene zurück (TransiEnt/Components/Electrical/Grid/Node.mo), da dieses Modell direkt verwendet wird?
- [ ] TransiEnt/Components/Electrical/Grid/Base/Node.mo: Größere Änderungen ohne Dokumentation
- [ ] TransiEnt/Components/Electrical/Prosumer/Check/Check_Prosumer.mo ist nicht  ausführbar und enthält keine Erläuterungen
- [ ] TransiEnt/Components/Electrical/Prosumer/Prosumer.mo ohne Dokumentation
- [ ] TransiEnt/Components/Sensors/Check/Check_SmartMeter.mo ohne Erläuterungen und nicht ausführbar. Könnte an Änderungen in TransiEnt/Components/Boundaries/Electrical/Base/PartialModelPowerBoundary.mo liegen, evtl. sind davon noch weitere Modelle betroffen, die ebefalls TransiEnt/Components/Boundaries/Electrical/Base/PartialModelPowerBoundary.mo nutzen
- [ ] Icons sind teilweise nicht im TransiEnt Design. Beispiele:
	- TransiEnt/Storage/Electrical/BEVBattery.mo
	- TransiEnt/Storage/Electrical/BatterySimple.mo
- [ ] TransiEnt/Storage/Base/GenericStorage_base.mo: Was war hier die Intention der Änderungen?
- [ ] TransiEnt/Consumer/Electrical/ElectricVehicle/Check/Check_BatteryElectricVehicle.mo nicht ausführbar

TSteffenTUHH committed:
Bugfixes and added documentation

wibela committed:
Document Basics.Blocks.Check.Check_OnOff

TSteffenTUHH committed:
Removed bak-mo files.

TSteffenTUHH committed:
Small model fixes for merge request.

TSteffenTUHH committed:
Reverted unintended change in TransiEnt/Components/Boundaries/Electrical/Base/PartialModelPowerBoundary.mo and replaced so position of the Node.mo.

TSteffenTUHH committed:
Changed documentation

TSteffenTUHH commented:
Moin @urju ,
die meisten deiner Anmerkungen konnte ich direkt in Änderungen umsetzen. Ich hab diese auch bereits angekreuzt. Ich hoffe man sieht das auf deiner Seite. 
Bezüglich der Examples/DistributionGrids habe ich die deutschen Kommentare denke ich gefunden. Was ich nicht verstanden habe, ist, warum die Check Modelle bei dir nicht laufen. Bei mir funktionieren diese einwandfrei und die jetzt hochgeladenen Änderungen sollten daran nichts verändert habe. 
Ruf dazu gerne nochmal an oder gibt mir hier eine Rückmeldung am besten mit einer Kopie der Fehlermeldung.

TSteffenTUHH committed:
Reverted new Complex Boundaries since they are not needed. Adjusted the new Models and Check Models for Prosumer, Heating, Heatpump, RoomHeating, DistributionGrids, PVPlanControllable, BatteryElectricVehicle and BatterySimple

TSteffenTUHH commented:
Aktuell noch offene Punkte:

- [ ] Line_new und dazugehöriges Partial Modell einpflegen.
- [ ] Heating, RoomHeating, Types, Node und DistributionGrids besser dokumentieren. Besonderes Augenmerk auf die Inputs und Outputs mit dem controlBus legen.
- [ ] Wird TransiEnt/Producer/Heat/Power2Heat/Heatpump/Controller/Base/HP_collector.mo gebraucht?
- [ ] Was sollen die Änderungen an TransiEnt/Producer/Electrical/Wind/Controller/TorqueController_modified.mo und TransiEnt/Storage/Base/GenericStorage_base.mo? -> Klären ob die Änderungen verworfen werden können.

TSteffenTUHH committed:
Revert PQBoundary.mo change

wibela committed:
Add documentation for building heating model

wibela committed:
Fix torque controller issues in Windturbine model

Affected model:
- TransiEnt.Producer.Electrical.Wind.Windturbine
- TransiEnt.Producer.Electrical.Wind.Controller.TorqueController

Observed behavior:
- Dynamic response is mostly as expected, but edge cases cause problems:
  1. At v_wind = 0 m/s, negative tau_friction leads to continuous,
     unbounded rotor acceleration independent of angular velocity.
  2. Rotor does not accelerate properly from standstill or at very low
     rotor speed.

Root causes:
- tau_n uses a constant gain independent of omega, resulting in
  unrealistic generator torque and power uptake from the grid
  ("ventilator mode").
- At low rotor speed (small omega), cp is very low, causing insufficient
  rotor torque.

Proposed solutions:
1. Invert the sign of tau_friction and scale it with a factor
   proportional to decreasing rotor speed (viscous friction model).
2. Use the relation λ_opt = ω_opt * R / v_wind to compare actual vs.
   optimal rotor speed. If deviation > 80%, switch off generator until
   deviation falls below 20%. Continuous variants may be preferable
   over discrete events, pending dynamic analysis.

Additionally:
- Consider reducing generator torque during rotor run-up.
- Further investigation needed on how this is handled in reality, since
  cp is very small at low tip-speed ratios.

wibela committed:
Remove HP_collector model

wibela commented:
Hier noch ein paar Anmerkungen:

Das Modell TransiEnt/Components/Electrical/Grid/Line_new.mo ist noch offen zur Änderung. Um das bestehende Modell TransiEnt.Components.Electrical.Grid.PiModelComplex beizubehalten müssten in den von @TSteffenTUHH erstellten Netzmodellen die Parameter geändert werden (`c` statt `b`, andere Bezeichnungen für den Strom, u.ä.). Hier wäre die Frage, ob du das mit den Skripten schneller ändern kannst, oder ob wir ein Convert Script schreiben.

Zum Heating:

> TransiEnt/Consumer/Heat/SpaceHeating/Check/Check_Heating.mo
> 
>     * Component type specifier Models_CyEntEE.CellModels.CPP.Boundaries.VDeltaBoundary not found, File: D:/Repos/Modelica/transient-lib/TransiEnt/Consumer/Heat/SpaceHeating/Check/Check_Heating.mo, line 6
> 
>     * Warum sind dort zusätzliche controlBuses?

Weil sonst der Expandable Connector nicht richtig erstellt wird im Modell (Connectoren sind nur mit Connectoren zu verbinden, aber ein Connector im Expandable Connector ist auf der ersten Ebene zunächst kein Connector, daher die Verschachtelung).

>     * Was machen step_T_buffer und step_T_room?

Waren dazu gedacht, die anderen Betriebsmodi zu testen.

>     * Erläuterungen, was man sich anschauen soll, wäre hilfreich

Das ist schwer zu sagen, da prinzipiell jede Variable von Interesse sein kann. Kann aber dazu schreiben, dass man sich die epp.P Variable und die Speichertemperatur anschauen soll.

>     * P_limit wird nicht sofort und nicht durchgängig eingehalten. Ist das so das beabsichtigte Modellverhalten?

Ja, es ist ja nur das Limit, welches nicht überschritten werden soll.

wibela committed:
Fix SpaceHeating.Heating PQ Boundary bug and add docs

wibela committed:
Heating LimitController Update (is now faster)

TSteffenTUHH committed:
Converted Line_new to PiModelComplexEfficient3ph, added Documentation and added a check model. AdmittanceTwoPort is deleted. Example Grids include the new line.

TSteffenTUHH committed:
Bugfixes in the MV models. Windturbine should now run.

wibela committed:
Heating LimitController Bug fix

TSteffenTUHH committed:
Improved documentation of the BEV characteristics

maggi94 committed:
Merge branch 'master' into EffiziEntEE
